lang/cs/Org.Apache.REEF.IMRU/API/IPerMapperConfigs.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Interface;
+
+namespace Org.Apache.REEF.IMRU.API
+{
+    /// <summary>
+    /// Class containing mapper specific configuration
+    /// </summary>
+    public interface IPerMapperConfigs
+    {
+        /// <summary>
+        /// Per-mapper configurations
+        /// </summary>
+        List<IConfiguration> MapperConfigurations { get; set; }
```

Comment:
Makes sense will change accordingly.

On Thu, Aug 27, 2015 at 5:28 PM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/API/IPerMapperConfigs.cs
> <https://github.com/apache/incubator-reef/pull/417#discussion_r38161921>:
>
> > +
> > +using System.Collections.Generic;
> > +using Org.Apache.REEF.Tang.Annotations;
> > +using Org.Apache.REEF.Tang.Interface;
> > +
> > +namespace Org.Apache.REEF.IMRU.API
> > +{
> > +    /// <summary>
> > +    /// Class containing mapper specific configuration
> > +    /// </summary>
> > +    public interface IPerMapperConfigs
> > +    {
> > +        /// <summary>
> > +        /// Per-mapper configurations
> > +        /// </summary>
> > +        List<IConfiguration> MapperConfigurations { get; set; }
>
> I am a bit uncomfortable with the partition id specific configuration as
> you suggested since this means that user knows the partition ids beforehand
> while writing the function mentioned above.
>
> Ah, I see. That makes sense.
>
> So one way is that instead of list of configuration we provide a function
> that takes as input number of mappers and returns set of configurations.
> This function can then be called by driver.
>
> A slight variation would be to ask for one configuration at a time,
> potentially telling the method the current partition number and the total
> number of partitions. That would be slightly more efficient in the case
> where there are only a small number of different configurations involved.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/417/files#r38161921>.
>